src/compiler/core.ts

```diff
@@ -2413,6 +2413,13 @@ namespace ts {
         return <T>(removeFileExtension(path) + newExtension);
     }
 
+    /**
+     * Takes a string like "jquery-min.4.2.3" and returns "jquery"
+     */
+    export function removeMinAndVersionNumbers(fileName: string) {
+        return fileName.replace(/((?:\.|-)min(?=\.|$))|((?:-|\.)\d+)/g, "");

```

Comment:
This also removes digits without the min, for example "jquery.4.2-test" also becomes "jquery-test". Probably not intended. (Also why use non-capturing groups (i.e. `(?:  )`? You are not using the groups for anything, so it's just syntactic noise).

Reply:
What exactly should the rule be? Remove all characters after the first . if the thing after the . is min or a number?

Reply:
@DanielRosenwasser wrote us a nice regex